### DISCUSSION TITLE
Getting error while pod install

### DISCUSSION THREAD

**Discussion by RNdevAvinash** [invertase/react-native-firebase#6151] [Q&A]
[!] CocoaPods could not find compatible versions for pod "Firebase/Analytics":
  In Podfile:
    RNFBAnalytics (from `../node_modules/@react-native-firebase/analytics`) was resolved to 14.7.0, which depends on
      Firebase/Analytics (= 8.14.0)

None of your spec sources contain a spec satisfying the dependency: `Firebase/Analytics (= 8.14.0)`.

You have either:
 * out-of-date source repos which you can update with `pod repo update` or with `pod install --repo-update`.
 * mistyped the name or version.
 * not added the source repo that hosts the Podspec to your Podfile.

firebase  versions 

    "@react-native-firebase/analytics": "^14.7.0",
    "@react-native-firebase/app": "^14.7.0",
    "@react-native-firebase/crashlytics": "^14.7.0",
    "@react-native-firebase/messaging": "^14.7.0",

**Comment by mikehardy** [comment]
I encourage you to use the pod-install package (`npx pod-install`), it will run `pod --repo-update` for you as needed like in situation such a this, then your spec sources will contain the new 8.14.0 release

**Reply by RNdevAvinash** [reply]
@mikehardy I tried with npx pod-install as well but getting same issue

**Reply by mikehardy** [reply]
Not sure what to say. The 8.14.0 pods exist, i can't reproduce this. Stack overflow is another good source of information for the error.

**Reply by Jackmekiss** [reply]
Same issue

**Reply by mikehardy** [reply]
I just ran this with 8.14.0 selected and it's fine https://github.com/mikehardy/rnfbdemo/blob/main/make-demo.sh
Same answers as before:

> I encourage you to use the pod-install package (npx pod-install), it will run pod --repo-update for you as needed like in situation such a this, then your spec sources will contain the new 8.14.0 release

and 

> i can't reproduce this. Stack overflow is another good source of information for the error.

**Comment by Gustav-Papinotas** [comment]
Try pod deintegrate then do pod install.

**Comment by pawan5209** [comment]
@RNdevAvinash Any solution you found for this.I am facing similar issues with @react-native-firebase/crashlytics. I just now upgraded react-native version from 0.63.4 to 0.67.4.

**Reply by mikehardy** [reply]
@pawan5209 if it is this error (similar wording maybe different versions)

> None of your spec sources contain a spec satisfying the dependency: Firebase/Analytics (= 8.14.0).

...you can check quite easily if the version *actually* exists, and adjust your troubleshooting accordingly

https://cocoapods.org/pods/Firebase (note the current version and the version in the podspec https://github.com/CocoaPods/Specs/blob/master/Specs/0/3/5/Firebase/8.15.0/Firebase.podspec.json)

Typically `npx pod-install` will auto-fix it for you or `pod repo update` will do it. But not if you some sort of strange caching in the way or your tools aren't working, that's all case by case

Hope this helps

**Reply by pawan5209** [reply]
@mikehardy What are the ways to ensure that no caching exists?I have already tried the whole step of 

-delete package-lock.json
-delete node_modules
-delete podfile-lock.json
-delete pod
-npm i
-npx pod-install

But the error persists.

**Reply by mikehardy** [reply]
`npx react-native-clean-project` is the gold standard, but every specific developer workstation on every specific project is its own unique challenge which you will have to figure out unfortunately - I won't be able to help further, the pods exist and you should be able to install them

**Comment by byyoungjin** [comment]
Maybe it would be related to the M1 mac book. 
When I install with npx pod-install, 
it fails with that dependency error and tries to update. 
but the update fails because of the M1 mac chip (arm64) 
if you get this error as a second error, when you update the pod, you should use rosetta terminal to install this. 
```
dlopen(/Library/Ruby/Gems/2.6.0/gems/ffi-1.15.5/lib/ffi_c.bundle, 0x0009): tried: '/Library/Ruby/Gems/2.6.0/gems/ffi-1.15.5/lib/ffi_c.bundle' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e')), '/usr/lib/ffi_c.bundle' (no such file) - /Library/Ruby/Gems/2.6.0/gems/ffi-1.15.5/lib/ffi_c.bundle
```

**Reply by mikehardy** [reply]
FFI has *still* not finished M1 mac compatibility? What a disaster. If not then you have to very very very very carefully kill all your pods then make sure that you only ever use pod command with the "arch" prefix setting it to x86_64. Basically you have to run everything under Rosetta2 at that point. FFI is a mess I guess.

**Comment by alexxsanchezm** [comment]
I don't know if this may help anyone, but I ran over very similar issue, this is what I did:

1.  Removed all my @react-native-firebase/[app, messaging, in-app-messaging, ...] which is my case.
2. Bump all my deps from @react-native-firabase/[app, messaging, in-app-messaging, ...] , all of them are on **14.12.0**.
3. Add to my Podfile: `pod 'Firebase/CoreOnly', '8.15.0'`, THIS WAS THE KEY, previous version of my rn-app didn't need to specify  any Firebase/CoreOnly version, I guess all deps used the same firebase core version. Previous Firebase/CoreOnly version was **8.8.8**.
4. Run `pod install --repo-update` and that's it.

Note: All @react-native-firebase/* libraries have more up to date versión(16.4.6). But I don't like to be always  on the latest cause is too risky for production apps, but use what fits your needs.

UPDATE: I did update all react-native-firebase dependencies to the latest version.

**Reply by mikehardy** [reply]
> Note: All @react-native-firebase/* libraries have more up to date versión(16.4.6). But I don't like to be always on the latest cause is too risky for production apps, but use what fits your needs.

I take specific issue with this statement. You should read the release notes to see what's actually in the releases. Bug fixes and compatibility with new mobile operating system versions. But if you want to stick with the buggy versions...okay :-)

**Reply by alexxsanchezm** [reply]
Yeah definitely, I was skeptical to update RNFB libs to the latest version(16.4.6).

**Comment by Ala3on** [comment]
Hi, I encountered this problem upgrading to version 16.4.6 and I couldn't find a way to fix it.I've tried everything and more, but I still get this error
````
[!] CocoaPods could not find compatible versions for pod "GoogleAppMeasurement":
  In Podfile:
    RNFBAnalytics (from `../node_modules/@react-native-firebase/analytics`) was resolved to 16.4.6, which depends on
      Firebase/Analytics (= 10.2.0) was resolved to 10.2.0, which depends on
        Firebase/Core (= 10.2.0) was resolved to 10.2.0, which depends on
          FirebaseAnalytics (~> 10.2.0) was resolved to 10.2.0, which depends on
            FirebaseAnalytics/AdIdSupport (= 10.2.0) was resolved to 10.2.0, which depends on
              GoogleAppMeasurement (= 10.2.0)

    ShopperBranding (= 0.3.0) was resolved to 0.3.0, which depends on
      Google-Mobile-Ads-SDK (~> 8) was resolved to 8.4.0, which depends on
        GoogleAppMeasurement (~> 7.0
```)`


any other ideas?

**Reply by mikehardy** [reply]
You will need to contact the vendor for this pod `ShopperBranding` and get them to update their Google-Mobile-Ads-SDK dependency to a more current one so that the GoogleAppMeasurement pod transitive dependency is compatible

Until that happens, if you must use whatever ShopperBranding is (I can't find any reference of it online...) you'll need to stay on whatever version of react-native-firebase supports a firebase-ios-sdk that uses GoogleAppMeasurement v7.x

Looks like that is anything that uses firebase-ios-sdk v7.x which is react-native-firebase v11.x

Those are so old I wouldn't use them at all, and can only strongly recommend not to attempt it, however `pod install` will work. https://github.com/invertase/react-native-firebase/blob/main/CHANGELOG.md#features-36

**Comment by diegomendezp** [comment]
I'm getting this error after install Firebase libraries:
`[!] Unable to find a specification for libevent (= 2.1.12)`

This is my Podfile:
```
source 'https://github.com/brightcove/BrightcoveSpecs.git'
require_relative '../node_modules/react-native/scripts/react_native_pods'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'

platform :ios, '12.4'
install! 'cocoapods', :deterministic_uuids => false

production = ENV["PRODUCTION"] == "1"

target 'caixaforum' do
  pod 'Brightcove-Player-Core/XCFramework'
  pod 'libevent', '2.1.12'
  config = use_native_modules!
  use_frameworks! :linkage => :static
  $RNFirebaseAsStaticFramework = true

  # Flags change depending on the env values.
  flags = get_default_flags()

  use_react_native!(
    :path => config[:reactNativePath],
    # Hermes is now enabled by default. Disable by setting this flag to false.
    # Upcoming versions of React Native may rely on get_default_flags(), but
    # we make it explicit here to aid in the React Native upgrade process.
    :hermes_enabled => true,
    :fabric_enabled => flags[:fabric_enabled],
    # Enables Flipper.
    #
    # Note that if you have use_frameworks! enabled, Flipper will not work and
    # you should disable the next line.
    :flipper_configuration => FlipperConfiguration.disabled,
    # An absolute path to your application root.
    :app_path => "#{Pod::Config.instance.installation_root}/.."
  )

  target 'caixaforumTests' do
    inherit! :complete
    # Pods for testing
  end

  post_install do |installer|
    react_native_post_install(installer, :mac_catalyst_enabled => false)
    __apply_Xcode_12_5_M1_post_install_workaround(installer)
    installer.pods_project.targets.each do |target|
      if (target.name.eql?('FBReactNativeSpec'))
        target.build_phases.each do |build_phase|
          if (build_phase.respond_to?(:name) && build_phase.name.eql?('[CP-User] Generate Specs'))
            target.build_phases.move(build_phase, 0)
          end
        end
      end
    end
    
    installer.pods_project.targets.each do |target|
      target.build_configurations.each do |config|
        config.build_settings["ENABLE_BITCODE"] = "NO"
      end
    end
    
    installer.pods_project.targets.each do |target|
      if target.respond_to?(:product_type) and target.product_type == "com.apple.product-type.bundle"
        target.build_configurations.each do |config|
            config.build_settings['CODE_SIGNING_ALLOWED'] = 'NO'
        end
      end
    end
  end
end

```

**Reply by mikehardy** [reply]
Maybe a network error? You may always search for cocoapods here, libevent @ 2.1.12 seems to be there https://cocoapods.org/pods/libevent

`pod repo update` maybe

**Reply by laurens-lamberts** [reply]
Probably you're no longer facing this issue given that it is a year ago, but I just faced the same issue and figured out the solution.
For anyone encountering the `libevent` Pod-installation message; this is due to the custom Cocoapods source specified in your Podfile. In the case of this thread that is `source 'https://github.com/brightcove/BrightcoveSpecs.git'`. The error occurs because Cocoapods is looking for all iOS Pods in your specified source now.

It is okay to specify your own Cocoapods source, but if you do so, make sure to include the official Cocoapods source as well.
You can do this by adding the following line above your own source;

```
source 'https://cdn.cocoapods.org/'
```

_Edit; updated source to use CDN as provided by mike hardy_

**Reply by mikehardy** [reply]
@laurens-lamberts that's a good pointer - only noting that cocoapods uses a CDN now that should be preferred as a standard source over the github.com source you mention: 

> source 'https://cdn.cocoapods.org/'

reference https://guides.cocoapods.org/syntax/podfile.html#source

**Reply by laurens-lamberts** [reply]
Hi @mikehardy, good to know, thanks! I will update our project immediately.
